network/tests/fuzzing.rs

```diff
@@ -137,3 +141,425 @@ async fn fuzzing_post_handshake() {
         node1.write_bytes(&random_bytes).await;
     }
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn fuzzing_corrupted_version_pre_handshake() {
+    // tracing_subscriber::fmt::init();
+
+    let node_setup = TestSetup {
+        consensus_setup: None,
+        ..Default::default()
+    };
+
+    let node = test_node(node_setup).await;
+    let node_addr = node.local_address().unwrap();
+
+    let mut rng = thread_rng();
+
+    for _ in 0..ITERATIONS {
+        let mut stream = TcpStream::connect(node_addr).await.unwrap();
+        let version = Version::serialize(&Version::new(1u64, stream.local_addr().unwrap().port())).unwrap();
+
+        // Replace a random percentage of random bytes at random indices in the serialised message.
+        let corrupted_version: Vec<u8> = version
+            .into_iter()
+            .map(|byte| if rng.gen_bool(0.1) { rng.gen() } else { byte })
```

Comment:
it would be nice if the alteration probability was a configurable `const` like `ITERATIONS`